src/qt/nametablemodel.h

```diff
@@ -0,0 +1,84 @@
+#ifndef NAMETABLEMODEL_H
+#define NAMETABLEMODEL_H
+
+#include "bitcoinunits.h"
+
+#include <QAbstractTableModel>
+#include <QStringList>
+
+class PlatformStyle;
+class NameTablePriv;
+class CWallet;
+class WalletModel;
+
+/**
+   Qt model for "Manage Names" page.
+ */
+class NameTableModel : public QAbstractTableModel
+{
+    Q_OBJECT
+
+public:
+    explicit NameTableModel(const PlatformStyle *platformStyle, CWallet* wallet, WalletModel *parent = 0);
+    //explicit NameTableModel(CWallet *wallet, WalletModel *parent = 0);
+    ~NameTableModel();
+
+    enum ColumnIndex {
+        Name = 0,
+        Value = 1,
+        ExpiresIn = 2
+    };
+
+    /** @name Methods overridden from QAbstractTableModel
+        @{*/
+    int rowCount(const QModelIndex &parent = QModelIndex()) const;
+    int columnCount(const QModelIndex &parent = QModelIndex()) const;
+    QVariant data(const QModelIndex &index, int role) const;
+    QVariant headerData(int section, Qt::Orientation orientation, int role) const;
+    QModelIndex index(int row, int column, const QModelIndex &parent = QModelIndex()) const;
+    Qt::ItemFlags flags(const QModelIndex &index) const;
+    /*@}*/
+
+private:
+    CWallet *wallet;
+    WalletModel *walletModel;
+    QStringList columns;
+    NameTablePriv *priv;
```

Comment:
This can be a `std::unique_ptr`, to simplify memory management.  (It would also avoid the memory leak you have due to the bug with double allocation, although the bug itself should still be fixed, of course.)